File: GestionAdherents/src/main/java/com/Spring/Bib/dao/GestionAdherents/Service/AdherentsService.java
package com.Spring.Bib.dao.GestionAdherents.Service;

import java.util.Collection;
import java.util.Optional;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import com.Spring.Bib.dao.GestionAdherents.Model.Adherents;
import com.Spring.Bib.dao.GestionAdherents.Repository.AdherentsRepository;


@Service("AdherentsService")
public class AdherentsService implements IAdherentsService {
	@Autowired
	private AdherentsRepository adherentsrepository;
	
	public Collection<Adherents> findAll() {

		return adherentsrepository.findAll();
	}

	
	public Adherents findone(String CIN) {
		
		return adherentsrepository.findById(CIN).orElse(null);
	}

	
	public void save(Adherents adherents) {
		
	adherentsrepository.save(adherents);
	}

	
	public void delete(String CIN) {
		adherentsrepository.deleteById(CIN);
	} 
public Adherents Update(Adherents adherent) {
		
		return adherentsrepository.save(adherent);
	}
	
}

File: bibliotheque/src/services/AdherentsService.js
import axios from "axios";

const ADHERENTS_API_BASE_URL ="http://localhost:8082/Gestion_Adherents";

class AdherentsService{
    saveAdherents(Adherents){
        return axios.post(ADHERENTS_API_BASE_URL,Adherents)
    }
    getAdherents(){
        return axios.get(ADHERENTS_API_BASE_URL)
    }
    DeleteAdherents(cin){
        return axios.delete(ADHERENTS_API_BASE_URL+"/"+cin)
    }
    getAdherentById(cin){
        return axios.get(ADHERENTS_API_BASE_URL+"/"+cin)
    }
    updateAdherents(Adherent,cin){
        return axios.put(ADHERENTS_API_BASE_URL+"/"+cin,Adherent)
    }
}
export default new AdherentsService();